### DISCUSSION TITLE
Distinguish between"not checked" and "not entered"

### DISCUSSION THREAD

**Discussion by disser** [iSoron/uhabits#96] [Feature Requests] [answered]
It would be useful to enter an X for a day deliberately. For habits such as "eat healthy," if you eat ice cream for breakfast you could immediately mark an X rather than remember at the end of the every thing that was eaten.

**Comment by iSoron** [comment]
Thank you for the suggestion. This feature makes sense, but I don't see how to add it while keeping the interface simple. I am open to suggestions.

One workaround for your scenario would be putting a check mark early in the morning and then, as soon as you fail the habit, you remove it. We could also allow check marks to appear automatically on new days.

**Comment by quark6789** [comment]
What about pressing and holding on the desired day (like we currently do to add check marks), then dragging down before releasing, to change the check to an X?  This would maintain the same gesture for marking checks, and wouldn't interfere with scrolling because you have to hold in one spot first.

**Comment by disser** [comment]
That sounds great, though I think it would have to be clear that only the selected check mark was being affected, perhaps by dropping a red X directly below the check mark. That would avoid confusion with affecting the entire goal. Also, there would have to be good separation between the regular check action and the X action, like make it an extra long hold?

**Comment by battaglr** [comment]
This is a very good idea!

**Comment by quark6789** [comment]
@disser I was imagining that adding a check mark would be the same gesture as it is now, with a hold and release on the desired spot, and making an X would be a hold, drag down, and release on the spot. Upon beginning the drag, the X would start to appear (either with a morph or fade) and it would phase in completely as the user continues to drag down (about half an inch). To cancel, the user just drags back up to where they started, and the X phases back out.

**Comment by aleksejrs** [comment]
I still fail every day to hold on the right spot to add a check mark, so when that happens, I drag up or down a little to avoid opening the habit's page.  Just saying, as a new touchscreen user.

**Comment by quark6789** [comment]
@aleksejrs Mm, good point.  But with what I'm proposing, you could just drag up, left, or right and still be fine.

**Comment by woj-tek** [comment]
A couple of suggestions (also related to the linked issue):
- habits could have tick (done), circle/square (pending), cross (not completed) states
- you could mark as done or not done from notification (3rd option to the current 'mark' and 'snooze')
- you could mark item on the list by long-pressing and it could pop-up a list with "done" or "ignored" (magine emoticons pop-up or alternative character keys on keyboards)

the third state could be optional so if not enabled the third option would not be present (no third option in notification, no pop-up on long press - simply marking as done currently)

**Comment by patcon** [comment]
fwiw, I agree that the simple interface is perhaps worth leaving this unaddressed :)

> One workaround [...] We could also allow check marks to appear automatically on new days.

Interesting, but I still think we'd be failing to capture important data that could be useful in visualization and analysis. (Particularly relevant to people interested in using this app for informal or formal studies of self.) I'm thinking about using this tool to publicize my own data in a visual way, to keep myself accountable.

Assuming fallible humans will sometimes forget, this work-around now makes it so that within the dataset, the "forget" data is less consistent -- sometimes forgetting will be represented as a success, and sometimes a failure :)

So it's an interesting idea if someone were only concerned with their user experience, but a bit less helpful for someone concerned about data completeness and integrity.

Thanks regardless of your final decision!

**Comment by battaglr** [comment]
@disser, @PunctualEmoticon, instead of creating a new gesture, we can do de following:
1. Long press: adds a success mark.
2. Long press over success mark: add a failure mark (not sure how this could look).
3. Long press over failure mark: returns to initial state.

In short, the same behaviour than now, but instead of just checking/unchecking, in the middle is the "failure" or "not completed" state.

What do you guys think?

**Comment by disser** [comment]
@battaglr I think this is the way to go.  If there can be a preference toggle for tri-state checkmarks, then the long-press behavior would be (gray check or X) -> (green check) -> (red X).  The default behavior would be (gray check or X) -> (green check).

**Comment by djibux** [comment]
I'm looking for this exact feature, and to me tri-state is the way to go.

I would do the following:
- Default state: a grey question mark
- Checked: a check mark 
- Failed: the X icon currently used in the interface

**Comment by djibux** [comment]
And by the way, that would also help differentiate between a habit you forgot to check and a habit you failed.

**Comment by woj-tek** [comment]
I simply love @djibux idea with x/?/v states ! I think it would be the best :-) Still - selecting failed / completed would be tricky...

**Comment by that0n3guy** [comment]
I want to +1 this feature.   I come from rewire and marking a habit as "failed" is a great.   If I just ignore it and don't put anything, there is no mental anguish, but if I have to say I failed something... there is a little bit of mental anguish and I would do better tomorrow.  Not only that, but **a bunch of red X's is motivating to stop slacking!**

I like @djibux idea but with a couple changes:
- Checked: a check mark
- Skipped: a grey checkmark: This is for non-everyday instances where the task/habit wasn't completed on that day
- Failed: a RED X icon.  Red checkmark to show its "bad"

Also, I think the following suggestions are **more confusing ways of doing it:**
- marking as failed with a long click then slide gesture - This would confuses the interface for new people and be easy to accidentally do the wrong thing, frustrating the user.
- Long press to mark "done", long press again to mark "failed".   Again confuses the user and take extra time to use the interface.

**I think a popup on single click with select options is a very simple way of doing it.**  I can do this in rewire in less time than a long click.    Its also less confusing for new people.

You can still keep the long press to mark "done".

Here is how rewire does it and its very fast to mark done/failed/skip:
![image](https://cloud.githubusercontent.com/assets/113870/18637257/7026436e-7e51-11e6-9b8a-ed87a1f4212e.png)

**Comment by that0n3guy** [comment]
Also, please notice how easy it is too add a note (issue: https://github.com/iSoron/uhabits/issues/56) for a day.   Its very fast.  You just click the "Tap to add a not for this day" section below the popup.

Adding a popup gives you more flexibility for other features (https://github.com/iSoron/uhabits/issues/56) while keeping the interface clean & simple.

**Comment by djibux** [comment]
A popup feels like a good solution to me.

Le 19 septembre 2016 17:18:42 GMT+02:00, that0n3guy notifications@github.com a écrit :

> Also, please notice how easy it is too add a note (issue:
> https://github.com/iSoron/uhabits/issues/56) for a day.   Its very
> fast.  You just click the "Tap to add a not for this day" section below
> the popup.
> 
> Adding a popup gives you more flexibility for other features
> (https://github.com/iSoron/uhabits/issues/56) while keeping the
> interface clean & simple.
> 
> ## 
> 
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/iSoron/uhabits/issues/96#issuecomment-248023748

## 

djib

**Comment by that0n3guy** [comment]
I played around with the code a little.   I've never messed w/ android coding before (we do web stuff at work) so it was a learning experience.   

I got as far as a simple mockup, none of the buttons work :) (and it looks crappy, rewire's layout is nicer).
![image](https://cloud.githubusercontent.com/assets/113870/19000378/d37b11ce-8707-11e6-94ea-ee971b875e7d.png)

I think I realized it would take me hours to figure this out so I probably won't continue, but I hope someone will implement this.   Its the single greatest feature I miss from rewire.

@iSoron It would be awesome if this could be marked as feature request instead of discussion :).

**Comment by adamvoss** [comment]
The simplicity of the interface was what tipped my decision to Loop and away from Rewire.  I, for one, am not a fan of needing a menu to mark every accomplishment.  There is a certain elegance and satisfaction so a simple tap making a check mark of accomplishment (I have short-press turned on).  It makes the app as quick and easy to use as tracking with pencil and paper.

With Rewire there is choosing the day then tracking to 4 or 5 different buttons (depending on the habit) to choose the action you want which creates extra mental load and requires reaching across the screen.

I would prefer to see a compact pop-up menu on press-and-hold.  Rather than how press-and-hold was discussed previously I would envision this working like many Android keyboards.  If you press and hold the menu will pop up with you already having the :heavy_check_mark:, but you would be able to slide your finger to release on one of the options, to choose them.  Much the way I press-and-hold on the `.` character to access the `?` character on the on-screen keyboard.  In this way the existing workflow would not be affected and additional options could be quickly accessed by the pop-up menu.

**Comment by djibux** [comment]
I like this! I definitely think the basic check-uncheck workflow should not be changed.

Le 2 octobre 2016 05:30:11 GMT+02:00, Adam Voss notifications@github.com a écrit :

> The simplicity of the interface was what tipped my decision to Loop and
> away from Rewire.  I, for one, am not a fan of needing a menu to mark
> every accomplishment.  There is a certain elegance and satisfaction so
> a simple tap making a check mark of accomplishment (I have short-press
> turned on).  It makes the app as quick and easy to use as tracking with
> pencil and paper.
> 
> With Rewire there is choosing the day then tracking to 4 or 5 different
> buttons (depending on the habit) to choose the action you want which
> creates extra mental load and requires reaching across the screen.
> 
> I would prefer to see a compact pop-up menu on press-and-hold.  Rather
> than how press-and-hold was discussed previously I would envision this
> working like many Android keyboards.  If you press and hold the menu
> will pop up with you already having the :heavy_check_mark:, but you
> would be able to slide your finger to release on one of the options, to
> choose them.  Much the way I press-and-hold on the `.` character to
> access the `?` character on the on-screen keyboard.  In this way the
> existing workflow would not be affected and additional options could be
> quickly accessed by the pop-up menu.
> 
> ## 
> 
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/iSoron/uhabits/issues/96#issuecomment-250951503

## 

djib

**Comment by that0n3guy** [comment]
> I would prefer to see a compact pop-up menu on press-and-hold. Rather than how press-and-hold was discussed previously I would envision this working like many Android keyboards

This is an interesting idea, but the tap-hold-slide-release still takes longer than tap (date) tap (check,fail, skip) button.

> I like this! I definitely think the basic check-uncheck workflow should not be changed.

I agree.    

## Enable/Disable In Settings

Also, I think that the submenu should have an "enable" option in the settings.   Something like `Enable sub-menu`.   HOW this submenu appears would depend on the state of of the `Toggle with short press` setting. 

Example:
- If `enable sub-menu` is checked and `toggle with short press` is UNchecked, the submenu appears on single click (like rewire)
- If `enable sub-menu` is checked and `toggle with short press` is CHECKED, the submenu appears on long click like @vossad01 suggested with the `I would envision this working like many Android keyboards`
- If `enable sub-menu` is UNchecked, the system acts as it does now.

The `enable sub-menu` setting would address the users who like the existing interface and don't want change.
The integration with the `toggle with short press` setting allows users to use the system the original way, while also allowing them to use the submenu on the fly if desired.

**Comment by disser** [comment]
I tried out Rewire (or whatever they call it today), and I don't like the pop-up menu. I think an option to enable tri-state checkmarks keeps the interface simple.

**Comment by adamvoss** [comment]
I would be fine with the tri-state.  The downside I saw to the tri-state is that it does not add a path for other features that have been requested namely notes (#56) and skip (#88).

**Comment by woj-tek** [comment]
> This is an interesting idea, but the tap-hold-slide-release still takes longer than tap (date) tap (check,fail, skip) button.

No necessarily - for example I've set swiftkey to distinguis long-tab as 200ms (a bit crazy, but something around 1s felt really slow ;) ).

**Comment by adamvoss** [comment]
> This is an interesting idea, but the tap-hold-slide-release still takes longer than tap (date) tap (check,fail, skip) button.

This also supposes removing the current press-and-hold workflow to check.  Under the current, default workflow all this adds is a small swipe as a final action.  I am not sure if maintaining this workflow if important.

Despite liking customization, I am a [little cautious of advocating for too many configuration options because I know it adds to software complexity](http://ometer.com/features.html).  However, if this is made configurable, I propose the following:
- Default behavior: Press and hold pops up the compact menu with check selected.  Release over the desired selection to choose.
- Short press: Perhaps(?) pop up the menu immediately, but otherwise behave as currently with the addition of the above menu behavior.  I am currently indifferent on whether the menu is immediate or after a hold delay in this case.
- Sticky menu: When you release after the menu appears, a choice will not have automatically been selected. You may then tap again to choose the desired option.

The above allows for preservation of all existing workflows while also allowing for a more persistent menu a la Remix.

What these configuration options do not allow that that0n3guy's does is for disabling the menu entirely.  I believe the menu should be reflective of the options for a given Habit and that the configuration should be per-habit rather than global.  If a habit does not allow for explicit failure, pass, etc., you would not see those items in the menu.  If the habit requires notes, you automatically take them to a popup that allows entering the notes and choosing the habit state.  In the event there is only one item in the menu perhaps you opt to not show the menu, though, I do not see that as a critical detail.

**Comment by that0n3guy** [comment]
@vossad01 I'm not sure your last paragraph makes much sense, or its over complicating the whole thing:

> that the configuration should be per-habit rather than global. If a habit does not allow for explicit failure, pass, etc., you would not see those items in the menu. If the habit requires notes, you automatically take them to a popup that allows entering the notes and choosing the habit state. In the event there is only one item in the menu perhaps you opt to not show the menu, though, I do not see that as a critical detail.

Currently all the settings are global (ie,  every habit can be checked and technically "failed", by means of non-action).   Your last paragraph seems to advocate that each habit could enable/disable the "fail" feature.  It also seems to talk about enabling/disabling notes on a per habit basis.   That seems overly complicated.   Adding a config per habit for those items seem overkill for most.   I think if simplicity is what your after, then the settings should be global.

On a side note, I'm not really too picky how the UI is, I was just trying to help.  I actually rarely use the UI and mostly just "check" the habits via the actionable notifications.

**Comment by adamvoss** [comment]
@that0n3guy I think the confusion is that I was writing with assumption that various current issues will be implemented and musing based on what the application would look like then.  Here is a little fuller explanation.

First, I suppose that habit-specific configuration related to entry will exist by the implementation Numerical Habits (#42) and thus could be expanded on by other features.  While you could make every task a numerical task to avoid this configuration, I cannot picture a way to do this that would seem very polished.

At present there are already two requests additional states: failure (#96) and pass (#88).  Both are tagged with `discussion` and my not be accepted for implementation.  I assume they are for the sake of discussion.  If they are generally viewed as useful states, their presence does not impede the workflow, and users can simply choose not to use them, then I do not see the need for a setting to able opt-in/opt-out.

However, a number of factors contribute to my thinking that maybe configuration is justified.  There does not appear to be a consensus on the need for these features so perhaps there is value in not allowing them on all habits. We already will have per-habit configuration so adding to that should not be too great of effort.  If these two additional states are added, we can not be sure these are the only ones that will be proposed or desired.  By allowing configuration per-habit the user's control of the behavior is maximized and support exists for adding more states in the future.

> It also seems to talk about enabling/disabling notes on a per habit basis

I realize now my use of 'requires notes' was ambiguous.  I meant that to mean that a user must be prompted to enter notes (i.e. the note is required), rather than a habit requires notes functionality.  Supposing habit completion notes are added (https://github.com/iSoron/uhabits/issues/56#issuecomment-226641475), I could see a user having habits(s) they always want to have notes, thus an option to require notes is a conceivable feature that would optimize the work flow for those habits.  I included this to show another way settings may influence the task completion UX, but probably should not have mentioned it.

NB. I am not advocating for the implementation of any or all of these.  I just trying to help out with how things may be implemented.  I have assumed so much will be implemented on observation that many of the issues create point towards implementing features available in commercial competitors to Loop and I figured that community interest would likely lead to their implementation.  @iSoron may have a more focused vision for Loop and I would support that.  Currently, these additional states are not ones I feel I need.

**Comment by that0n3guy** [comment]
@vossad01 I gotcha.   I never saw the numerical issue https://github.com/iSoron/uhabits/issues/42 and think that causes other issues and turns this app into something like a lifelogging app.

> If these two additional states are added

I think this is still confusing, because these states already exist, see my "back on topic" section below.

We have somewhat derailed this issue, **mostly my fault I think.**   Lets try to keep the discussion of the other issues (#42, #88, #56) in there respective places.

### Back on topic - Overview and refocus

Based on what this app does (IE, simply/easily tracks if habbits take place).   There are 3 states: Pass (checked), Fail (unchecked), skip (used only in non-everyday habits).

All of these states are already in the current version:
- Pass(checked) is done manually by the user
- Fail(unchecked) is done automatically when the user does nothing
- skip happens automatically when you do nothing in a certain timeframe.

(the above is how the system currently handles the states)

**To me, this discussion is about the "fail" state, and if we should allow manually marking it as failed.** 

I think manually marking a habit as failed, doesn't do anything to the apps functionality (it doesn't change streaks, habit strength, history).   Its just a visual indication.  Manually marking failed and automatic marking as failed are the same, they just look different.

**So I think the simplest way to implement this features is to allow a global setting for "Allow manually marking a habit failed"**

**Comment by nj999** [comment]
Hello.

I just joined github to thank you for your app :) and to add my comments.

I am also looking for this feature. We want to mark certain habits that we have failed today. So the simplest implementation will be to have 3 status - 

pending(? - grey color)
done(check/tick - green color)
failed(x - red color)

In the beginning all rows will be ?/grey. once you tap on any one of them it will turn to done/green. if you tap again on it will turn to fail/red. if you tap on it then it will be pending/grey. i think this will keep the simplicity of the app without any pop-ups or complex logic. I dont think we will mind tapping two times to mark a row as failed. Some folks have said to allow skip. I dont know why you would want to skip, but if you want to allow skip then allow 4th status for skip and let them continue with tapping to go to their status.

Also, keep in mind that there are certain habits which are automatically checked/ticked because it was once a week and we have done it in this week. They are currently shown as grey tick. You should continue that as it is. Do not convert them to pending when we start a new day. Only make pending what is really pending for today.

Please consider adding "fail" status as soon as you can because I have close to 100 rows now and I want to simplify my view by marking few things as failed so that i know what are left to finish for today.

**Comment by edjroot** [comment]
Hello! Any news on this? :)

**Comment by fiosmiles** [comment]
This would be very useful!

**Comment by Erudition** [comment]
First tap: Miss
Second tap: Check 

Simple. Then the default workflow is to double tap. That's one idea.

Either way, it's not necessary to have the user distinguish between "skip" and "fail". They need only a "miss" option (which is automatically assumed at the end of a day with no check). The software can easily know whether that miss is a fail or a skip! 

After all, if you have a habit to be done at least once every three days, you may not be happy about missing it on the second day (which is perhaps when you intended to complete it), but it's not yet a failure - you still have another day! So, it gets marked as a skip. On the third day, the miss would go down as a fail. (Along with subsequent consecutive misses.)

On the other hand, if a habit is to be done every day, every miss is a fail! Why make the user specify? 😊

Then in the UI:
Check: ✅
Skip:    ➖
Fail:     ❌

**Comment by imanjundi** [comment]
i think this feature as @disser requested "enter an X for a day deliberately" is really important because it would make it easy to see which habits are still relevant for the day. 

**Why it's important?**
if you already know in the morning that you missed your morning jogging, it's a bit annoying and distracting for your brain to see this habit with no indication throughout the whole day! As @nj999 said, **marking failed habits as soon as possible**, makes it clear and easy to **focus throughout the day on the habits that are actually still open** without being distracted by habits that you already know you failed.

**Comment by mgttlinger** [comment]
There is also the need for a third state to mark a habit "not applicable" for the day which I guess is distinct from "failed" or "not entered" in the sense that it shouldn't count towards the metrics.

As an example, I have a goal to take the stairs to my office instead of the elevator which I want to do every workday. Now, if I am working from home or if I am on a business trip or vacation I obviously can't check this in general but it doesn't mean I failed the goal to do this every day.

**Comment by Erudition** [comment]
I'd like to re-emphasize that I think "_missed_" is a really good word to describe the third state.

@mgttlinger If you must view your habit as a _success_ or _failure_, you might want to reframe what your habit is - in this case, your desire is to _avoid_ taking the stairs. Now you have a question you can answer every day (habits ideally do not depend on the environment) - "did you avoid taking the stairs?" - and if you simply did not go to work, the answer is "yes!". Think about it: would you expect a sugar addict to mark a day as a 'failure' if they had a day where there simply was no sugar around? No. The temptation does not need to be there all the time for you to successfully avoid it.

Of course, it's better to have a positive habit than a negative one (but now it's getting into psychology) - if the point is the _exercise_ of taking the stairs, you should have a fallback exercise you can do on days when taking the stairs is not applicable. :)

**Comment by adrienbeau** [comment]
> habits ideally do not depend on the environment

In an ideal world maybe. Practically speaking, they do, and in many case, it is fine to skip some days, as long as you fall back to your habit after that. Case in point: I measure my weight every day at home, but I do not when I am away (for work or holiday). I don't consider the days away as a success or failure, they are unrelated to the habit (currently my score tanks badly every time I am away).

Besides, I have several work habits that do not apply to my off time, especially not my holidays. There's no *reframing* around that. Not doing a work-related habit on holiday is neither a success nor a failure. Actually, many of my habits do not apply during my holidays, and I think it is important to be able to take a break from habits too.

A good test of a habit may be the ability to fall back into it after a holiday. It would certainly be easier with scores that have not tanked for lack of applicability.

**Comment by Erudition** [comment]
> In an ideal world maybe. Practically speaking, they do

Maybe I should have phrased that differently. It's not the world that needs to be ideal, it's the habit. I gave an example of a practical way the habit can apply to every day. There aren't any practical obstacles to overcome. There may however be philosophical ones.

> A good test of a habit may be the ability to fall back into it after a holiday

Actually, that is more a test of willpower that particular day, so statistically, with many habits it's nearly a random chance (depending on the habit's difficulty). While on your break from your habit, you have already broken it - so there isn't much to test anyway (especially if you've broken the Never Miss Twice rule).

> in many case, it is fine to skip some days
>  don't consider the days away as a success or failure, they are unrelated to the habit (currently my score tanks badly every time I am away).
> not doing a work-related habit on holiday is neither a success nor a failure

I remember thinking this way as well. Perhaps you are right. Either way, I clearly won't be able to convince you. I stopped thinking of habits like this after lots of reading, though, so maybe you'd be open to hearing from the experts. There are lots of great resources, including the new & renowned _Atomic Habits_, or if books aren't your style, there's quick videos on topics such as why [the best habit is the one you do _every_ day](https://www.youtube.com/watch?v=dxrRCb-7A1I), or there's articles that summarize the latest expertise in this field as well. For example, this [How To Make Good Habits Stick](https://liveboldandbloom.com/12/habits/how-to-make-good-habits-stick) article succinctly describes major results from research such as the need for an "implementation intention":

> **Create an “implementation intention.”** Reframe your habit as an “If/ Then” statement. For example, “If I’ve finished my breakfast, then I’ll do five pushups.”

This is right along the lines of the framing I presented earlier, like "If I'm presented with the opportunity to take the stairs or an elevator, I'll take the stairs". Again, if the lack of stairs on your break/holiday/vacation really bothers you, you could always add a similar replacement action for those days that currently don't apply. But again this is otherwise not strictly necessary -- just as a sugar addict gets to call every day without sugar a "win", even if there are some days when the cravings do not occur. Agree or not, If you treat your habit this way, you will never have to deal with that "tanking" problem!

> and I think it is important to 
> take a break from habits too.

Fascinating opinion - do you have any particular evidence for this belief? I'd love to believe that as well. Though It seems to fly in the face of everything we know about habits, kinda akin to one's dentist saying "it's important to be able to take a break from proper dental hygiene". It certainly would be nice to hear, though!

**Comment by r4taw2mg** [comment]
As this topic comes up on a regular base, I would like to know the opinion of @iSoron  : Do you think about implementing something in this direction?

**Comment by c33s** [comment]
> One workaround for your scenario would be putting a check mark early in the morning and then, as soon as you fail the habit, you remove it. We could also allow check marks to appear automatically on new days.

@iSoron https://github.com/iSoron/uhabits/issues/96#issuecomment-218694446

i really like the idea of auto adding check marks. i am currently eating vegan, i do this all the time, so i have to check each day that i only ate vegan. sometimes i notice that some food was not vegan as i thought, so i would remove the checkmark manually.  the auto check would be so helpful. just few days where you not have done as you do normally.

the interface would be easy on habbit creation just add a checkbox "auto-check" and the interface is done.

**Comment by iSoron** [comment, marked as answer]
Fixed in aa288ac4061238, and should become available in Loop 2.0. For now, to enter a lapse, you will need to toggle the checkmark twice. If skips are enabled, you will need to toggle it three times. Mocks and pull requests to make this more streamlined are welcome.

**Reply by lukaszlew** [reply]
Is the release of 2.0 as beta on playstore blocked by [Milestone 11](https://github.com/iSoron/uhabits/milestone/11) ?
If so, that's a rather large feature while 2.0 has a lot of quite useful features already (like the distinction between "not checked" and "not entered").

Is there a way to release it sooner?

**Comment by hiqua** [comment]
@iSoron as of now, this feature is not implemented for cases where the entries which the user was not expected to tick. For instance, if you have two habits that should be done every 3 days in this state:

![2021-08-08-214411_466x480_scrot](https://user-images.githubusercontent.com/3812042/128643982-f1d34cce-5c5f-423b-b032-29fdce6bbdad.png)

Then ticking THU 5 will result in the same representation for FRI 6:

![2021-08-08-214421_477x354_scrot](https://user-images.githubusercontent.com/3812042/128643984-dc9a0c65-545d-424d-ae1d-adc0969c372c.png)

I'm still going through the code, but from what I've seen so far, I think that:
* the data in the DB does contain the NO (which is why it works fine when unticking THU 5)
* but the values in CheckmarkButtonView.kt are the ones given by some interval computation which replaces the NO by YES_AUTO where appropriate

In a way, I feel that these should not be the same types: NO / YES / UNKNOWN are really specific to one specific checkmark, while YES_AUTO depends on the surrounding values. For now, I can probably figure out a quick solution but just computing pairs of values (e.g. (NO, YES_AUTO), (UNKNOWN, NO) etc., where the first value is what the user entered, and the second is the result of the computation), but I'll most likely find a cleaner way eventually.

**Reply by iSoron** [reply]
@hiqua In the current version of the app, `YES_AUTO` means *"you did not perform the habit, but a repetition was not expected for that date,"* while `NO` means *"you did not perform the habit, and a repetition WAS expected for that date"*; so I think that the current app behavior is correct; it is impossible to get a `NO` checkmark for Friday 6, in your example, because a repetition is not expected for that day.

This feature request was originally describing a different issue, which we have already addressed.

**Reply by hiqua** [reply]
Ok, I'll open a new discussion.
